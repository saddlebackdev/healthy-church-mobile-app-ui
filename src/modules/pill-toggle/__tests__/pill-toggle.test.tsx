// npx jest ./src/modules/pill-toggle/__tests__/pill-toggle.test.ts

import * as React from 'react';
import {fireEvent, render} from '@testing-library/react-native';

import PillToggle from '../pill-toggle';
import ThemeProvider from '../../theming/theme-provider';
import defaultTheme from '../../theming/default-theme';

describe('PillToggle', () => {
  let onPressMock: any, options: any;

  beforeEach(() => {
    onPressMock = jest.fn();

    options = [
      {
        label: 'Selected',
        testID: 'pill-toggle-1',
        accessibilityLabel: 'pill-toggle-1',
        value: 1,
        onPress: onPressMock,
      },
      {
        label: 'Unselected',
        testID: 'pill-toggle-2',
        accessibilityLabel: 'pill-toggle-2',
        value: 2,
        onPress: onPressMock,
        disabled: true,
      },
      {
        label: 'Disabled',
        testID: 'pill-toggle-3',
        accessibilityLabel: 'pill-toggle-3',
        value: 3,
        onPress: onPressMock,
      },
    ];
  });

  describe('Rendering', () => {
    it('renders itself correctly', () => {
      const wrapper = render(
        <ThemeProvider theme={defaultTheme}>
          <PillToggle selected={1} options={options} />
        </ThemeProvider>,
      );

      const pillToggle = wrapper.getByTestId('pill-toggle');

      expect(pillToggle.children).toHaveLength(options.length);

      expect(pillToggle).toBeDefined();

      expect(wrapper).toMatchSnapshot();
    });

    it('renders the children correctly', () => {
      const {getByTestId} = render(
        <ThemeProvider theme={defaultTheme}>
          <PillToggle selected={1} options={options} />
        </ThemeProvider>,
      );

      const option1 = getByTestId(options[0].testID);
      const option2 = getByTestId(options[1].testID);
      const option3 = getByTestId(options[2].testID);

      expect(option1).toBeDefined();
      expect(option2).toBeDefined();
      expect(option3).toBeDefined();
    });
  });

  describe('Interactions', () => {
    it('should call onPress function on click of pill button', () => {
      const {getByTestId} = render(
        <ThemeProvider theme={defaultTheme}>
          <PillToggle selected={1} options={options} />
        </ThemeProvider>,
      );

      const button = getByTestId('pill-toggle-1');

      fireEvent.press(button);

      expect(onPressMock).toHaveBeenCalledTimes(1);
    });

    it('should not call onPress function on click of pill button when option is disabled', () => {
      const {getByTestId} = render(
        <ThemeProvider theme={defaultTheme}>
          <PillToggle selected={1} options={options} />
        </ThemeProvider>,
      );

      const button = getByTestId('pill-toggle-2');

      fireEvent.press(button);

      expect(onPressMock).not.toHaveBeenCalled();
    });
  });
});
