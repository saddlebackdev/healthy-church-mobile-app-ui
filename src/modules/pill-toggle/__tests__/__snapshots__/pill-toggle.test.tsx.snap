// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PillToggle Rendering renders itself correctly 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
        "flexDirection": "row",
      },
    ]
  }
  testID="pill-toggle"
>
  <View
    accessibilityLabel="pill-toggle-1"
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#00AEFF",
        "borderBottomLeftRadius": 22,
        "borderBottomRightRadius": 0,
        "borderColor": "#00AEFF",
        "borderRightWidth": 0,
        "borderStyle": "solid",
        "borderTopLeftRadius": 22,
        "borderTopRightRadius": 0,
        "borderWidth": 1,
        "opacity": 1,
      }
    }
    testID="pill-toggle-1"
  >
    <Text
      accessibilityLabel="pill-toggle-1-label"
      inversed={true}
      muted={false}
      small={true}
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontFamily": "System",
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "600",
            "paddingHorizontal": 22,
            "paddingVertical": 10,
            "textAlign": "left",
          },
        ]
      }
      testID="pill-toggle-1-label"
      weight="semiBold"
    >
      Selected
    </Text>
  </View>
  <View
    accessibilityLabel="pill-toggle-2"
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#EDF1F5",
        "borderBottomLeftRadius": 0,
        "borderBottomRightRadius": 0,
        "borderColor": "#DBE0E3",
        "borderRightWidth": 0,
        "borderStyle": "solid",
        "borderTopLeftRadius": 0,
        "borderTopRightRadius": 0,
        "borderWidth": 1,
        "opacity": 1,
      }
    }
    testID="pill-toggle-2"
  >
    <Text
      accessibilityLabel="pill-toggle-2-label"
      inversed={false}
      muted={true}
      small={true}
      style={
        Array [
          Object {
            "color": "#97A4AB",
            "fontFamily": "System",
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "600",
            "paddingHorizontal": 22,
            "paddingVertical": 10,
            "textAlign": "left",
          },
        ]
      }
      testID="pill-toggle-2-label"
      weight="semiBold"
    >
      Unselected
    </Text>
  </View>
  <View
    accessibilityLabel="pill-toggle-3"
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "borderBottomLeftRadius": 0,
        "borderBottomRightRadius": 22,
        "borderColor": "#DBE0E3",
        "borderRightWidth": 1,
        "borderStyle": "solid",
        "borderTopLeftRadius": 0,
        "borderTopRightRadius": 22,
        "borderWidth": 1,
        "opacity": 1,
      }
    }
    testID="pill-toggle-3"
  >
    <Text
      accessibilityLabel="pill-toggle-3-label"
      inversed={false}
      muted={true}
      small={true}
      style={
        Array [
          Object {
            "color": "#97A4AB",
            "fontFamily": "System",
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "600",
            "paddingHorizontal": 22,
            "paddingVertical": 10,
            "textAlign": "left",
          },
        ]
      }
      testID="pill-toggle-3-label"
      weight="semiBold"
    >
      Disabled
    </Text>
  </View>
</View>
`;
